1109-i386-txt-Add-Intel-TXT-definitions-header-file.patch

```diff
@@ -0,0 +1,712 @@
++
++struct grub_txt_heap_madt_element
++{
++  /* Copy of ACPI MADT table */
++} GRUB_PACKED;
++
++#define GRUB_TXT_HEAP_EXTDATA_TYPE_EVENT_LOG_POINTER2_1	8
++
++struct grub_txt_heap_event_log_pointer2_1_element
++{
++  grub_uint32_t type;
++  grub_uint32_t size;
++  grub_uint64_t phys_addr;
++  grub_uint32_t allocated_event_container_size;
++  grub_uint32_t first_record_offset;
++  grub_uint32_t next_record_offset;
++} GRUB_PACKED;
++
++#define GRUB_TXT_HEAP_EXTDATA_TYPE_MCFG			9
++
++struct grub_txt_heap_mcfg_element
++{
++  /* Copy of ACPI MCFG table */
++} GRUB_PACKED;
++
++/* TXT Heap Tables */
++
++struct grub_txt_bios_data
++{
++  grub_uint32_t version; /* Currently 5 for TPM 1.2 and 6 for TPM 2.0 */
++  grub_uint32_t bios_sinit_size;
++  grub_uint64_t reserved1;
++  grub_uint64_t reserved22;
++  grub_uint32_t num_logical_procs;
++  /* Versions >= 5 with updates in version 6 */
++  grub_uint32_t sinit_flags;
++  grub_uint32_t mle_flags;
++  /* Versions >= 4 */
++  /* Ext Data Elements */
++} GRUB_PACKED;
++
++/* GRUB SLAUNCH specific definitions OS-MLE data */
++#define GRUB_SL_BOOTPARAMS_OFFSET	0x12c
++#define GRUB_SL_MAX_EVENT_LOG_SIZE	(5*4*1024)   /* 4k*5 */
++#define GRUB_SL_MAX_VARIABLE_MTRRS	32
++#define GRUB_SL_OS_MLE_STRUCT_VERSION	1
++
++struct grub_slaunch_mtrr_pair
++{
++  grub_uint64_t mtrr_physbase;
++  grub_uint64_t mtrr_physmask;
++} GRUB_PACKED;
++
++struct grub_slaunch_mtrr_state
++{
++  grub_uint64_t default_mem_type;
++  grub_uint64_t mtrr_vcnt;
++  struct grub_slaunch_mtrr_pair mtrr_pair[GRUB_SL_MAX_VARIABLE_MTRRS];
++} GRUB_PACKED;
++
++struct grub_txt_os_mle_data
++{
++  grub_uint32_t version;
++  grub_uint32_t boot_params_addr;
++  grub_uint64_t saved_misc_enable_msr;
++  struct grub_slaunch_mtrr_state saved_bsp_mtrrs;
++  grub_uint32_t ap_wake_block;
++  grub_uint32_t ap_wake_block_size;
++  grub_uint64_t evtlog_addr;
++  grub_uint32_t evtlog_size;
++  grub_uint8_t mle_scratch[64];
++} GRUB_PACKED;
++
++struct grub_txt_os_sinit_data
++{
++  grub_uint32_t version; /* Currently 6 for TPM 1.2 and 7 for TPM 2.0 */
++  grub_uint32_t flags;
++  grub_uint64_t mle_ptab;
++  grub_uint64_t mle_size;
++  grub_uint64_t mle_hdr_base;
++  grub_uint64_t vtd_pmr_lo_base;
++  grub_uint64_t vtd_pmr_lo_size;
++  grub_uint64_t vtd_pmr_hi_base;
++  grub_uint64_t vtd_pmr_hi_size;
++  grub_uint64_t lcp_po_base;
++  grub_uint64_t lcp_po_size;
++  grub_uint32_t capabilities;
++  /* Version = 5 */
++  grub_uint64_t    efi_rsdt_ptr;
++  /* Versions >= 6 */
++  /* Ext Data Elements */
++  grub_uint8_t ext_data_elts[];
++} GRUB_PACKED;
++
++struct grub_txt_sinit_mle_data
++{
++  grub_uint32_t version;             /* Current values are 6 through 9 */
++  /* Versions <= 8 */
++  grub_uint8_t bios_acm_id[20];
++  grub_uint32_t edx_senter_flags;
++  grub_uint64_t mseg_valid;
++  grub_uint8_t sinit_hash[20];
++  grub_uint8_t mle_hash[20];
++  grub_uint8_t stm_hash[20];
++  grub_uint8_t lcp_policy_hash[20];
++  grub_uint32_t lcp_policy_control;
```

Comment:
What are these fields for version 9?  Are they zero?  Are they skipped entirely, in which case this needs to be a union?